components/header-bar/src/profile/profile-menu.js

```diff
@@ -69,6 +69,7 @@ const ProfileContents = ({ name, email, avatarUrl, helpUrl }) => {
                             baseUrl,
                             'dhis-web-commons-security/logout.action'
                         )}
+                        onClick={onLogout}
```

Comment:
@KaiVandivier are we sure this will fire and complete (could take some time if the cache is large) before the redirect to the logout action?

Reply:
I was just realizing that -- no I don't think there's a guarantee that this will happen before the logout action. 

If this app is hosted on the same domain as the rest of the instance, I think that's okay though -- CacheStorage and IndexedDB can be accessed domain-wide

Reply:
Still I think it would be good to wait to navigate until the `onLogout` function is complete to make this more robust

Reply:
The `MenuItem` prevents the default, so I don't think any navigation would happen if the `onLogout` callback is passed: https://github.com/dhis2/ui/blob/ee4005bd83cdabb4896444f05cd2ceedf3ef8e9d/components/menu/src/menu-item/menu-item.js#L13-L27